### ISSUE TITLE
Porting WebServer to rust with Pyo3

### ISSUE THREAD

**Issue by aka-deVOID** [AliRn76/panther#18]
I'm working on a new feature for Panther, I port the webserver to Rust.
with async support. to enhance the performance ...

for this thing I use [Hyper](https://github.com/hyperium/hyper) crate + TokIO for async handling ...

please, do not close this issue ... I will add PR soon.

**Comment by aka-deVOID** [comment]
@AliRn76
if you have any idea or do you want to move part of the process to the Rust web server tell me here.

**Comment by AliRn76** [comment]
This is a good idea, go for it, we are going to discuss this soon 👌

**Comment by AliRn76** [comment]
Do you have something like https://github.com/litmus-web/litmus in your mind?

**Comment by aka-deVOID** [comment]
> Do you have something like https://github.com/litmus-web/litmus in your mind?

yes, but top of another framework, we need to improve speed and request handling and move some layers to the web server ...
this is precisely what can make Panther better than another framework, I'm working on it.
I will keep you informed of the changes here ...

**Comment by aka-deVOID** [comment]
but I have a request, to be a project maintainer after RP.

**Comment by he0xA1** [comment]
PR is up: <#152>

The port is working end to end — HTTP and websockets both — on hyper + tokio via
PyO3, with uvicorn kept as the default and the Rust server opt-in behind
`panther run --server rust`.

Design note: Rust never calls into the interpreter on its own. It publishes accepted
connections onto a bounded queue that the asyncio loop drains through
`Server.accept()`, so every entry into Python happens on the loop thread. That keeps
the GIL off the accept path.

Verified against CPython 3.14 on arm64 macOS, with a CI job covering fmt, clippy and
the test suite on Linux.

Still open: HTTP/2 and TLS aren't implemented, and I haven't benchmarked it against
uvicorn yet.
